backend/app/models/user.rb

```diff
@@ -36,4 +36,27 @@ class User < ApplicationRecord
   delegate :admin?, to: :role
   delegate :teacher?, to: :role
   delegate :student?, to: :role
+
+  after_create :slack_notify
+
+  private
+
+  def slack_notify
```

Comment:
If we were going with this approach I would suggest a more descriptive name like `slack_notify_after_creation`